### PR TITLE
275 import simulation json

### PR DESCRIPTION
**Pull request by msevestre**

### CODE REVIEW

**Comment by msevestre** [comment]
By far the most difficult mapper so far (and probably overall)

File: src/PKSim.Core/Model/OutputSchemaFactory.cs [hunk 1]
```diff
@@ -17,20 +16,17 @@ public class OutputSchemaFactory : IOutputSchemaFactory
...
+      public OutputSchemaFactory(IObjectBaseFactory objectBaseFactory, IOutputIntervalFactory outputIntervalFactory, IDimensionRepository dimensionRepository)
       {
          _objectBaseFactory = objectBaseFactory;
          _outputIntervalFactory = outputIntervalFactory;
          _dimensionRepository = dimensionRepository;
-         _displayUnitRetriever = displayUnitRetriever;
       }
 
-      public OutputSchema Create()
+      public OutputSchema CreateEmpty()
```
Comment (msevestre) [inline src/PKSim.Core/Model/OutputSchemaFactory.cs]: This was never used by I actually need one EMPTY. So change implementation

File: src/PKSim.Core/Model/Simulation.cs [hunk 2]
```diff
@@ -588,12 +592,20 @@ public virtual ModelProperties ModelProperties
       /// <summary>
       ///    Returns the <see cref="OSPSuite.Core.Domain.OutputSchema" /> defined in the settings
       /// </summary>
-      public virtual OutputSchema OutputSchema => SimulationSettings.OutputSchema;
+      public virtual OutputSchema OutputSchema
+      {
+         get => SimulationSettings.OutputSchema;
+         set => SimulationSettings.OutputSchema = value;
```
Comment (msevestre) [inline src/PKSim.Core/Model/Simulation.cs]: Setter required everywhere

File: src/PKSim.Core/Snapshots/Mappers/CompoundPropertiesMapper.cs [hunk 3]
```diff
@@ -73,6 +84,29 @@ private CompoundProcessSelection compoundProcessSelectionFor(IReactionMapping re
          return compoundProcessSelection;
       }
 
+      private IReactionMapping reactionMappingFrom(CompoundProcessSelection compoundProcessSelection, Model.CompoundProcess process)
+      {
+         switch (process)
```
Comment (msevestre) [inline src/PKSim.Core/Snapshots/Mappers/CompoundPropertiesMapper.cs]: Really only trying to save the very minimal info required to build the simulation from scratch

File: src/PKSim.Presentation/Services/ProjectTask.cs [hunk 4]
```diff
@@ -231,7 +231,7 @@ public void LoadProjectFromSnapshot()
 
          async Task ProjectFromSnapshot()
```
Comment (msevestre) [inline src/PKSim.Presentation/Services/ProjectTask.cs]: This works great here. However the HeavyWorkManager does not support async stuff propertly. We will have to update that as well

File: src/PKSim.Core/Snapshots/Mappers/OutputSchemaMapper.cs [hunk 5]
```diff
@@ -10,10 +11,14 @@ namespace PKSim.Core.Snapshots.Mappers
    public class OutputSchemaMapper : SnapshotMapperBase<ModelOutputSchema, SnapshotOutputSchema>
    {
       private readonly OutputIntervalMapper _outputIntervalMapper;
+      private readonly IOutputSchemaFactory _outputSchemaFactory;
+      private readonly IContainerTask _containerTask;
 
-      public OutputSchemaMapper(OutputIntervalMapper outputIntervalMapper)
+      public OutputSchemaMapper(OutputIntervalMapper outputIntervalMapper, IOutputSchemaFactory outputSchemaFactory, IContainerTask _containerTask)
```
Comment (rwmcintosh) [inline src/PKSim.Core/Snapshots/Mappers/OutputSchemaMapper.cs]: Is this supposed to be _containerTask or containerTask?

File: src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs [hunk 6]
```diff
@@ -122,9 +116,115 @@ private string[] usedObervedDataFrom(ModelSimulation simulation)
...
+         var compounds = compoundsFrom(snapshot.Compounds, project);
+         var modelProperties = modelPropertiesFrom(snapshot.Model, simulationSubject);
+
+         var simulation = _simulationFactory.CreateFrom(simulationSubject, compounds, modelProperties);
+         MapSnapshotPropertiesToModel(snapshot, simulation);
+
+         await mapCompoundProperties(snapshot.Compounds, project, simulation);
+         simulation.EventProperties = await _eventPropertiesMapper.MapToModel(snapshot.Events, project);
+
+         //Once all used building blocks have been set, we need to ensure that they are also synchronized in the in the simulation
```
Comment (rwmcintosh) [inline src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs]: Change that comment to remove the duplicated 'in the in the'

File: src/PKSim.Core/Snapshots/Mappers/OutputSchemaMapper.cs (hunk 5 above)
Comment (msevestre) [reply src/PKSim.Core/Snapshots/Mappers/OutputSchemaMapper.cs]: containerTask

File: src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs (hunk 6 above)
Comment (msevestre) [reply src/PKSim.Core/Snapshots/Mappers/SimulationMapper.cs]: done

File: src/PKSim.Core/Snapshots/Mappers/OutputSchemaMapper.cs (hunk 5 above)
Comment (msevestre) [reply src/PKSim.Core/Snapshots/Mappers/OutputSchemaMapper.cs]: done

**Pull request merged**